Bs 59 websocket msg

sahabaishakhi opened the pull request.

sahabaishakhi committed:
add basic build for ws demo #59

sahabaishakhi committed:
changed asset path for discord #59

sahabaishakhi committed:
messages not getting updated/received in dom #59

sahabaishakhi committed:
Merge branch 'master' of https://github.com/224-1/xtnt-micro into bs-59-websocket-msg

sahabaishakhi committed:
adding untap condition #59

sahabaishakhi committed:
message format changed from hashmap for put #59

sahabaishakhi committed:
error for client to server sending/receiving msgs #59

sahabaishakhi committed:
simple message passing implementation using websockets #59

sahabaishakhi committed:
websocket receive not working #59

sahabaishakhi committed:
websocket receive not working #59

azureus89 committed:
Simple Websocket Example
- Removed chord from dependencies (was using deprecated functions in http-kit/
old version "2.1.19" - main cause of inconsistency in send/recieve behaviour)
- removed files / lines of code that were hence redundant
- Removed core.async based implementation (should integrate later after
benchmarking standard websocket implementation - reducing immediate complexity)
- BREAKING UPDATE! Commented out current chat handler - this will break chat
example on "/" route in dev build
- Created basic websocket handler example (ws-handler) in handler.clj
- TODO reimplement chat handler on top of current example in future issues
- TODO add documentation for extending the websocket handler for use in real
world applications (data format required from client - and event handler functions to be implemented on the server)
- TODO cljs client implementation to show interaction with websocket

sahabaishakhi committed:
Merge pull request #61 from 224-1/az-59-websocket-error

Simple Websocket Example

sahabaishakhi committed:
Merge branch 'bs-59-websocket-msg' of https://github.com/224-1/xtnt-micro into bs-59-websocket-msg

sahabaishakhi committed:
build error failing to load resource #59

sahabaishakhi committed:
haslett implementation error in different build #59

sahabaishakhi committed:
haslett implementation client side #59

sahabaishakhi committed:
websocket message passing using haslett #59

sahabaishakhi committed:
documentation websocket haslett #59

sahabaishakhi committed:
websocket test url #59

azureus89 commented:
@sahabaishakhi 
1) websocket use example needs to be a bit more practical ... show a text box and a send button, and render messages recieved into p tags ... remember when we started we talked about not really chat but show messages being printed into DOM, and a text box and send button to show its working? The user gets nothing from the page as it is right now, its just something for you to be feel good about - has no explanatory / exploratory value for the user.

2) Documentation is fine for now. But you can be very good at this, so in the future try to be more succint. Think about the reader, Eliminate anything that is unnecessary and doesn't benefit the user. For starters separate the "guide" for the end user from the documentation of how server side works - this part is not important for the cljs end user - they are simply going to trust that you/the server is going to do their job as message broker reliably. Review all documentation, and bring it down the basics, keep the files to a minimum, lot of past work is now redundant, and documentation has been made redundant with it.

which leads me to 
3) xtnt-micro can now be only one build, delete everything else, chat, trello, discord. They are built using techniques / frameworks we are going to be leaving behind. Keep only the simple websocket handler on the server side + one line instruction to run the server (go to shell, do lein run - thats it) .. and the websocket example.
A small amount of documentation for the server working is required, extensive detail not necessary, its too simple an implementation at this stage for it .. no mult / no async / single app. Just keep this separate, and let users know where to go for what.
Use the main project readme for project high level details + instructions on clj / cljs operation. And keep the server side (short) documentation in docs, and only accessible if the user desires.

Go ahead and do all this on this branch, no need for new issue. Try to finish by EOD.

sahabaishakhi committed:
build name correction in doc #59

sahabaishakhi committed:
clojurescript update dom error #59

azureus89 committed:
Working websocket implementation with reagent and go-loop
- Recieved messages parsed to edn
- Unique msg ids for reagent
- Messages rendered on dom
- go-loop macro used instead of go block
- organised code regions to separate concerns
- notes for issue completion

sahabaishakhi committed:
sending message via textbox not working #59

sahabaishakhi committed:
updation of dom from text-box upon hot-reload #59

sahabaishakhi committed:
reagent form not showing keys typed in input box #59

sahabaishakhi committed:
unable to use swap! on persistentArrayMap external state atom #59

sahabaishakhi committed:
form-2 using local state to update global state error #59

sahabaishakhi committed:
other hiccup element not working along with form-2 #59

sahabaishakhi committed:
rendering text-box and dom updation using ws go-loop and recur #59

sahabaishakhi committed:
using core async channel for receiving messages #59

sahabaishakhi committed:
client side core async usage issue #59

azureus89 commented:
Keeping this PR open is an insult to your intelligence ... im closing this, please re-open when done
